libvips/foreign/heifsave.c

```diff
@@ -545,6 +545,13 @@ vips_foreign_save_heif_build( VipsObject *object )
 	heif->page_height = vips_image_get_page_height( heif->image );
 	heif->n_pages = heif->image->Ysize / heif->page_height;
 
+	if( (guint64) heif->page_width * heif->page_height > INT_MAX / 4 ||
+		heif->page_width > SHRT_MAX ||
+		heif->page_height > SHRT_MAX ) {
```

Comment:
`SHRT_MAX` (=32767) is not equivalent to the previous 32768 dimension limit, or was this intentional?

Reply:
Ha, not intentional but I suppose equally arbitrary. The more I think about it, the more I prefer 16384 as a maximum edge as that's the libvips default input limit.

Reply:
I've reduced the max edge to 16384.